features/lorawan/LoRaWANInterface.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file
+ *
+ * @brief      Implementation of LoRaWANBase
+ *
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "lorawan/LoRaWANInterface.h"
+
+inline LoRaWANStack& stk_obj()
```

Comment:
Return should be a pointer, not a reference as NetworkStack does.

Reply:
The reference makes sure that we have only one instance and it cannot be NULL. 
For consistency it would make sense otherwise I don't see a difference. I wanted to avoid Null checks, if its a reference I can make it const and ensure that it is being passed around correctly otherwise throw me a compilation error. With pointer that is not possible.